yt_dlp/extractor/youtube.py

```diff
@@ -3307,7 +3306,57 @@ def _extract_heatmap(self, data):
                 'value': ('intensityScoreNormalized', {float_or_none}),
             })) or None
 
-    def _extract_comment(self, comment_renderer, parent=None):
+    def _extract_comment(self, view_model, entities, parent=None):
+        comment_entity_payload = traverse_obj(entities, (..., 'payload', 'commentEntityPayload', {dict}), get_all=False)
+        toolbar_entity_payload = traverse_obj(entities, (..., 'payload', 'engagementToolbarStateEntityPayload', {dict}), get_all=False)
+        comment_id = comment_entity_payload.get('properties').get('commentId')
```

Comment:
- use the same "early exit `if not comment_id`" pattern from the old method
- use `get_first`, which is equivalent to `traverse_obj(obj, (..., path), get_all=False)`
- never use `.get` chains, use traversal

```suggestion
        comment_entity_payload = get_first(entities, ('payload', 'commentEntityPayload', {dict}))
        toolbar_entity_payload = get_first(entities, ('payload', 'engagementToolbarStateEntityPayload', {dict})) or {}
        comment_id = traverse_obj(comment_entity_payload, ('properties', 'commentId', {str}))
        if not comment_id:
            return
```